modules/cfe_testcase/src/time_external_test.c

```diff
@@ -0,0 +1,88 @@
+/*************************************************************************
+**
+**      GSC-18128-1, "Core Flight Executive Version 6.7"
+**
+**      Copyright (c) 2006-2019 United States Government as represented by
+**      the Administrator of the National Aeronautics and Space Administration.
+**      All Rights Reserved.
+**
+**      Licensed under the Apache License, Version 2.0 (the "License");
+**      you may not use this file except in compliance with the License.
+**      You may obtain a copy of the License at
+**
+**        http://www.apache.org/licenses/LICENSE-2.0
+**
+**      Unless required by applicable law or agreed to in writing, software
+**      distributed under the License is distributed on an "AS IS" BASIS,
+**      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+**      See the License for the specific language governing permissions and
+**      limitations under the License.
+**
+** File: time_external_test.c
+**
+** Purpose:
+**   Functional test of basic External Time Source APIs
+**
+**   Demonstration of how to register and use the UT assert functions.
+**
+*************************************************************************/
+
+/*
+ * Includes
+ */
+
+#include "cfe_test.h"
+
+int32 TestCallbackFunction(void)
+{
+    return CFE_SUCCESS;
+}
+
+int32 TestCallbackFunction2(void)
+{
+    return CFE_SUCCESS;
+}
+
+void TestExternal(void)
+{
+#if ((CFE_PLATFORM_TIME_CFG_SRC_MET == true) || (CFE_PLATFORM_TIME_CFG_SRC_GPS == true) || \
+     (CFE_PLATFORM_TIME_CFG_SRC_TIME == true))
+    CFE_TIME_SysTime_t time = {1000, 0};
+#endif
+
+    UtPrintf("Testing: CFE_TIME_ExternalTone, CFE_TIME_ExternalMET, CFE_TIME_ExternalGPS, CFE_TIME_ExternalTime");
+
```

Comment:
Worth a UtPrintf or comment here that the following time calls could impact the system timekeeping.  Likely impact is incorrect time for one update cycle, a rejected external time update, multiple tone's or external updates detected, or similar.  The call here confirms the implementation is properly included, more extensive timekeeping verification is a project/system responsibility.